src/storage/txn/scheduler.rs

```diff
@@ -485,6 +534,15 @@ impl Scheduler {
         self.register_report_tick(event_loop);
     }
 
+    /// Event handler for new command.
+    ///
+    /// This method will try to acquire all the necessary latches. If successful, initiates a get
+    /// snapshot for furthur processing; otherwise, adds the command to the waiting queue(s), it
+    /// will be handled later in `wakeup_cmd` when its turn comes.
+    ///
+    /// Note that once a command is ready to execute, the snapshot is always up-to-date during the
+    /// execution because 1) all the conflicting commands (if any) must be in the waiting queues;
+    /// 2) there may be non-conflicitng commands running concurrently, but it doesn't matter.
```

Comment:
For readonly commands, we do not ensure the snapshot is "up-to-date". For instance, a Prewrite is followed by a Get. Since Get do not need to acqire any latches, its snapshot may be before or after Prewrite' execution.
Actually, the latch is more for making sure no write could be lost(or partially lost) when they are parallel executing in different threads.